GliaWidgetsTests/SecureConversations/Welcome/SecureConversations.WelcomeViewModelSpec.swift

```diff
@@ -0,0 +1,196 @@
+import Foundation
+import XCTest
+@testable import GliaWidgets
+
+final class SecureConversationsWelcomeViewModelTests: XCTestCase {
+    var viewModel: SecureConversations.WelcomeViewModel = .mock
+
+    override func setUp() {
+        viewModel = .mock
+    }
+
+    func testDelegateGetsCalledOnBackTapped() {
+        var isCalled = false
+
+        viewModel.delegate = { event in
+            switch event {
+            case .backTapped:
+                isCalled = true
+            default: XCTFail()
+            }
+        }
+
+        viewModel.event(.backTapped)
+
+        XCTAssertEqual(isCalled, true)
+    }
+
+    func testDelegateGetsCalledOnCloseTapped() {
+        var isCalled = false
+
+        viewModel.delegate = { event in
+            switch event {
+            case .closeTapped:
+                isCalled = true
+            default: XCTFail()
+            }
+        }
+
+        viewModel.event(.closeTapped)
+
+        XCTAssertEqual(isCalled, true)
+    }
+}
+
+// Is attachment available
+extension SecureConversationsWelcomeViewModelTests {
+    func testIsAttachmentAvailable() {
+        var environment: SecureConversations.WelcomeViewModel.Environment = .mock
+        environment.fetchSiteConfigurations = { completion in
+            let site: CoreSdkClient.Site = try! .mock(
+                allowedFileSenders: .init(operator: true, visitor: true)
+            )
+
+            completion(.success(site))
+        }
+
+        viewModel = .init(environment: environment, availability: .mock)
+
+        XCTAssertTrue(viewModel.isAttachmentsAvailable)
+    }
+
+    func testIsAttachmentNotAvailable() {
```

Comment:
You can mark you tests to be throwing, and thus avoiding usage of `try!`. This is better because in case of throwing error your whole test suite will not crash, letting other tests to be evaluated, while throwing test will just fail.

```suggestion
    func testIsAttachmentAvailable() throws {
         var environment: SecureConversations.WelcomeViewModel.Environment = .mock
         environment.fetchSiteConfigurations = { completion in
             let site: CoreSdkClient.Site = try .mock(
                 allowedFileSenders: .init(operator: true, visitor: true)
             )

             completion(.success(site))
         }

         viewModel = .init(environment: environment, availability: .mock)

         XCTAssertTrue(viewModel.isAttachmentsAvailable)
     }
```